docs/reference/ccr/index.asciidoc

```diff
@@ -2,28 +2,300 @@
+comes online, replication resumes between the clusters.
+
+image::images/ccr-arch-bi-directional.png[Bi-directional configuration where each cluster contains both a leader index and follower indices]
+
+NOTE: This configuration is useful for index-only workloads, where no updates
+to document values occur. In this configuration, documents indexed by {es} are
+immutable. Clients are located in each datacenter alongside the {es}
+cluster, and do not communicate with clusters in different datacenters.
+
+[discrete]
+[[ccr-data-locality]]
+==== Data locality
+Bringing data closer to your users or application server can reduce latency
+and response time. This methodology also applies when replicating data in {es}.
+For example, you can replicate a product catalog or reference dataset to 20 or
+more datacenters around the world to minimize the distance between the data and
+the application server.
+
+In the following diagram, data is replicated from one datacenter to three
+additional datacenters, each in their own region. The central datacenter
+contains the leader index, and the additional datacenters contain follower
+indices that replicate data in that particular region. This configuration
+puts data closer to the application accessing it.
+
+image::images/ccr-arch-data-locality.png[A centralized datacenter replicated across three other datacenters, each in their own region]
+
+[discrete]
+[[ccr-centralized-reporting]]
+==== Centralized reporting
+Using a centralized reporting cluster is useful when querying across a large
+network is inefficient. In this configuration, you replicate data from many
+smaller clusters to the centralized reporting cluster.
+
+For example, a large global bank might have 100 {es} clusters around the world
+that are distributed across different regions for each bank branch. Using
+{ccr}, the bank can replicate events from all 100 banks to a central cluster to
+analyze and aggregate events locally for reporting. Rather than maintaining a
+mirrored cluster, the bank can use {ccr} to replicate specific indices.
+
+In the following diagram, data from three datacenters in different regions is
+replicated to a centralized reporting cluster. This configuration enables you
+to copy data from regional hubs to a central cluster, where you can run all
+reports locally.
+
+image::images/ccr-arch-central-reporting.png[Three clusters in different regions sending data to a centralized reporting cluster for analysis]
+
+[discrete]
+[[ccr-replication-mechanics]]
+=== Replication mechanics
+Although you <<ccr-getting-started,set up {ccr}>> at the index level, {es}
+achieves replication at the shard level. When a follower index is created,
+each shard in that index pulls changes form its corresponding shard in the
+leader index, which means that a follower index has the same number of
+shards as its leader index. All operations are replicated by the follower, so that operations to create, update, or delete a document are replicated.
```

Comment:
> All operations are replicated by the follower, so that operations to create, update, or delete a document are replicated.`

I am confused about this.

Reply:
@dnhatn, that sentence is confusing. How about this version instead?
>All operations on the leader are replicated by the follower, such as operations to create, update, or delete a document.

Reply:
Should it be "replicated **to** the follower" instead of "replicated **by** the follower"?

Reply:
Hmm 🤔 Isn't the follower index the one that is doing the replicating? 

I could go either way on this one if you feel more strongly about the revised text, or feel that it is more accurate.

Reply:
I am fine with the current.